scripts/images/convert_qcow2_ova.py

```diff
@@ -115,7 +251,96 @@ def create_tar(image_file_source, image_file_path):
+    mount_dir = tmpdir + '/' + 'tempMount'
+    rhel_bash_file = mount_dir  + '/' + 'rhel_bash.sh'
+    rhel_cloud_config_file = mount_dir  + '/etc/cloud/' + 'cloud.cfg'
+    real_root = os.open("/", os.O_RDONLY)
+    os.mkdir(mount_dir) # Temporary mount directory
+    print("Getting a free loop device ...")
+    cmd = 'losetup --nooverlap  --partscan  -f ' + extracted_raw_file_path
+    out, err, ret = exec_cmd(cmd)
+    if ret != 0:
+        print('ERROR: Failed to get a free loop device:', err)
+        sys.exit(2)
+    cmd = 'losetup -l --output NAME,BACK-FILE | grep ' + extracted_raw_file_path + '| awk \'{print $1}\' '
+    out, err, ret = exec_cmd(cmd)
+    if ret != 0:
+        print('ERROR: Getting loop device name failed:', err)
+        sys.exit(2)
+
+    loop_device = out.rstrip()
+
+    print("probing partition table ...")
+    cmd = 'partprobe ' + loop_device
+    out, err, ret = exec_cmd(cmd)
+    if ret != 0:
+        print('ERROR: Getting loop device name failed:', err)
+        sys.exit(2)
+
+    try:
+        print("mounting the loop device ...")
+        cmd = 'mount ' + loop_device + 'p2' + ' ' + mount_dir
+        out, err, ret = exec_cmd(cmd)
+        if ret != 0:
+            print('ERROR: Failed mounting the device:', err)
+            sys.exit(2)
+        for sdir in ('/proc', '/dev', '/sys', '/var/run/'):
+            cmd = 'mount -o bind ' + sdir + ' ' + mount_dir + sdir
+            out, err, ret = exec_cmd(cmd)
+            if ret != 0:
+                print('ERROR: Failed mounting the device:', err)
+                sys.exit(2)
+
+        rhel_bash_template = Template(template_rhel_bash)
+        rhel_bash = rhel_bash_template.render(rh_sub_username=rhUser, rh_sub_password=rhPassword, root_password=osPassword )
+        with open(rhel_bash_file, "w") as stream:
+            stream.write(rhel_bash)
+        st = os.stat(rhel_bash_file)
+        os.chmod(rhel_bash_file, st.st_mode | stat.S_IEXEC)
+
+        rhel_cloud_config_template = Template(template_cloud_config)
+        rhel_cloud_config = rhel_cloud_config_template.render()
+        with open(rhel_cloud_config_file, "w") as stream:
+            stream.write(rhel_cloud_config)
+
+        os.chroot(mount_dir)
+        os.chdir('/')
+
+        print("Running script ...")
+        cmd = './rhel_bash.sh'
+        out, err, ret = exec_cmd(cmd)
+        print("shell script :", out)
+        if ret != 0:
+            print('ERROR: failed to run the script:', err)
+            sys.exit(2)
+
+    finally:
+        os.fchdir(real_root)
+        os.chroot('.')
+        os.close(real_root)
+        print("Unmounting all")
+        for sdir in ('/proc', '/dev', '/sys', '/var/run/'):
+                cmd = 'umount ' + mount_dir + sdir
+                out, err, ret = exec_cmd(cmd)
+                if ret != 0:
+                        print('ERROR: Failed mounting the device:', err)
+                        sys.exit(2)
```

Comment:
I see exit call only here, not sure if we need a code exit here for unmounting failure!